### PR TITLE
Adding custom resource quota for vault deployment in smaug

### PR DESCRIPTION
**Pull request by dystewart**
Adding custom resource quota to reflect the storage needs of the vault deployment.

### CODE REVIEW

File: cluster-scope/base/core/namespaces/vault/resourcequota.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+kind: ResourceQuota
+apiVersion: v1
+metadata:
+  name: vault-custom
+spec:
+  hard:
+    requests.cpu: '500'
```
Comment (HumairAK) [inline cluster-scope/base/core/namespaces/vault/resourcequota.yaml:7]: 500 cores?!? did you mean `500m` as in millicores?

Comment (dystewart) [reply cluster-scope/base/core/namespaces/vault/resourcequota.yaml:7]: Yikes yeah 500 cores is probably overkill... updating to millicores

**Comment by HumairAK** [comment]
@dystewart can you resolve the build errors

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve

**Pull request merged**